lib/ansible/module_utils/facts.py

```diff
@@ -1621,6 +1623,58 @@ def get_dmi_facts(self):
             if found:
                 self.facts['product_name'] = found.group(1)
 
+    def get_device_facts(self):
+        self.facts['devices'] = {}
+
+        disk_stats = {
+            'Product': 'product',
+            'Revision': 'revision',
+            'Serial No': 'serial',
+            'Size': 'size',
+            'Vendor': 'vendor',
+            'Hard Errors': 'hard_errors',
+            'Soft Errors': 'soft_errors',
+            'Transport Errors': 'transport_errors',
+            'Media Error': 'media_errors',
+            'Predictive Failure Analysis': 'predictive_failure_analysis',
+            'Illegal Request': 'illegal_request',
+        }
+
+        cmd = ['/usr/bin/kstat', '-p']
+
+        for ds in disk_stats:
+            cmd.append('sderr:::%s' % ds)
+
+        d = {}
+        rc, out, err = self.module.run_command(cmd)
+        if rc != 0:
+            return dict()
+
+        sd_instances = set(sorted([line.split(':')[1] for line in out.split('\n') if line.startswith('sderr')]))
+        for instance in sd_instances:
+            lines = (line for line in out.split('\n') if len(line) > 0 and line.split(':')[1] == instance)
```

Comment:
I don't know how sane the data you're reading in is but this might be a little safer:

``` python
lines = (line for line in out.split('\n') if ':' in line and line.split(':')[1] == instance)